lib/tasks/images.rake

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+namespace :images do
+  # Images can be lost for several reasons, for example:
+  #
+  # - file loss on disk
+  # - failed upload
+  # - file loss on bucket
+  # - file loss when changing storage
+  #
+  # If we can't recover the image, we should delete the record in the database
+  # as well. Otherwise operations like regenerating thumbnails will try to
+  # find those images every time which costs time and bandwidth.
+  task delete_orphans: :environment do
+    puts "To identify orphaned images, update the metadata first:"
+    puts ""
+    puts "  bundle exec rake paperclip:refresh:metadata CLASS=Spree::Image"
+    puts ""
+
+    orphaned_images = Spree::Image.where(attachment_file_size: 0)
```

Comment:
The issue is that 

```
irb(main):001:0> image = Spree::Image.find_by(id: 101)
irb(main):008:0> image.attachment_file_size
=> 9249
```
so we should instead check

```
irb(main):007:0> image.attachment.exists?
=> false
```
to get the right count. Obviously, that will be very slow as it'll do a TCP connection for each file.

Reply:
nice finding!
I think we can combine both and go for it?